### PR TITLE
Sleeper server basic concept

### PR DESCRIPTION
**Pull request by qwertyquerty**
## About The Pull Request

Creates a config flag that toggles a "sleeper server" run mode where most subsystems are disabled. The goal is to have a lightweight version of the server that exists only to respond to topic requests for the API.

## Why It's Good For The Game

Will let us process things in byond that need to be processed in byond for the API without lagging the game servers, like rendering character sprites, for instance.

## Testing Photographs and Procedure

Normally, a Beestation server idle's around `866.7MB` after startup. This config flag brings it down to `344.8MB` and uses a negligible CPU amount.

I have a feeling we can make it even more lightweight so this is a work in progress.

## Changelog
:cl:
server: Create lightweight sleeper server run mode
/:cl:

### CODE REVIEW

**Comment by qwertyquerty** [comment]
If we don't load centcom, we can shave it down to around `337.5MB` but centcom is included at compile time so we couldn't do it with just this config flag

**Comment by AffectedArc07** [comment]
Have you considered reeimplementing the BYOND world/Topic protocol (which is known at this point) in some other application to avoid needing to run DD at all?

File: code/controllers/subsystem/dcs.dm [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PROCESSING_SUBSYSTEM_DEF(dcs)
 	name = "Datum Component System"
-	flags = SS_NO_INIT
+	flags = SS_NO_INIT | SS_SLEEPER
```
Comment (AffectedArc07) [inline code/controllers/subsystem/dcs.dm]: Also qwerty my guy

File: code/controllers/subsystem/fail2topic.dm [hunk 2]
```diff
@@ -1,7 +1,7 @@
 SUBSYSTEM_DEF(fail2topic)
 	name = "Fail2Topic"
 	init_order = INIT_ORDER_FAIL2TOPIC
-	flags = SS_BACKGROUND
+	flags = SS_BACKGROUND || SS_SLEEPER
```
Comment (AffectedArc07) [inline code/controllers/subsystem/fail2topic.dm]: The lack of consistency in flag application

File: code/controllers/subsystem/garbage.dm [hunk 3]
```diff
@@ -2,7 +2,7 @@ SUBSYSTEM_DEF(garbage)
 	name = "Garbage"
 	priority = FIRE_PRIORITY_GARBAGE
 	wait = 2 SECONDS
-	flags = SS_POST_FIRE_TIMING|SS_BACKGROUND|SS_NO_INIT
+	flags = SS_POST_FIRE_TIMING|SS_BACKGROUND|SS_NO_INIT|SS_SLEEPER
```
Comment (AffectedArc07) [inline code/controllers/subsystem/garbage.dm:5]: Is colossal

**Comment by qwertyquerty** [comment]
> Have you considered reeimplementing the BYOND world/Topic protocol (which is known at this point) in some other application to avoid needing to run DD at all?

We need to run DD because we need the game to generate character sprites. That's the entire point of what this is for

File: code/__DEFINES/MC.dm [hunk 4]
```diff
@@ -57,6 +57,9 @@
 /// This flag overrides SS_KEEP_TIMING
 #define SS_POST_FIRE_TIMING 64
 
+/** Subsystem is still initialized in the lightweight sleeper-server mode */
+#define SS_SLEEPER 128
```
Comment (PowerfulBacon) [inline code/__DEFINES/MC.dm:61]: This name seems confusing to me, sleeper sounds like it would sleep if it was a sleeper server

**Review by PowerfulBacon — COMMENTED** [review]
I feel like this could be done a lot more efficiently with some program written in another language dedicated to doing this, running along side the server and fetching code/sprites from the game server files

**Comment by qwertyquerty** [comment]
> I feel like this could be done a lot more efficiently with some program written in another language dedicated to doing this, running along side the server and fetching code/sprites from the game server files

It would take too much work to maintain

**Review by PowerfulBacon — APPROVED** [review]
Seems fine but I think if we were actually going to generate sprite previews for the website then it would need some external program just pulling the games code

**Comment by PowerfulBacon** [comment]
If we get something that uses this and there is no better solution, reopen it. Otherwise its stale and unlikely to get merged wtihout a need for it for now

**Pull request closed without merging**